src/domain/relation/establish.ts

```diff
@@ -0,0 +1,16 @@
+
+import type Requester from '../authentication/Requester';
+import type RelationData from './data/RelationData';
+import create from './data/create';
+import exists from './data/exists';
+import RelationAlreadyExists from './errors/RelationAlreadyExists';
+
+export default async function establish(requester: Requester, followingId: string): Promise<RelationData>
+{
+    if (await exists(requester.id, followingId))
```

Comment:
```suggestion
    const relationExists = await exists(requester.id, followingId);
    
    if (relationExists)
```
We try to avoid `await` in `if` statements.